accounts/abi/bind/backends/simulated.go

```diff
@@ -95,7 +95,10 @@ func NewSimulatedBackendWithDatabase(database ethdb.Database, alloc core.Genesis
 	backend.filterSystem = filters.NewFilterSystem(filterBackend, filters.Config{})
 	backend.events = filters.NewEventSystem(backend.filterSystem, false)
 
-	backend.rollback(blockchain.CurrentBlock())
+	header := backend.blockchain.CurrentBlock()
+	block := backend.blockchain.GetBlock(header.Hash(), header.Number.Uint64())
```

Comment:
Theoretically we should check nilness here, but should be fine in simulation backend.

Reply:
Yeah, I did try to make sure thing won't end up in production code, but for tests I didn't bother. The simulator is a bit interesting in that it's "production but test code", however I don't expect people to concurrently modify the simulator while running it, so IMHO should be fine.